jmespath-vertx/src/main/java/io/burt/jmespath/vertx/VertxRuntime.java

```diff
@@ -0,0 +1,213 @@
+package io.burt.jmespath.vertx;
+
+import io.burt.jmespath.BaseRuntime;
+import io.burt.jmespath.JmesPathType;
+import io.burt.jmespath.RuntimeConfiguration;
+import io.burt.jmespath.jcf.JcfRuntime;
+import io.vertx.core.json.Json;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+import java.util.*;
+
+import static io.burt.jmespath.JmesPathType.*;
+
+public class VertxRuntime extends JcfRuntime {
+    public VertxRuntime() {
+        this(RuntimeConfiguration.defaultConfiguration());
+    }
+
+    public VertxRuntime(RuntimeConfiguration configuration) {
+        super(configuration);
+    }
+
+    @Override
+    public Object parseString(String str) {
+        return Json.decodeValue(str);
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public List<Object> toList(Object value) {
+        if (value instanceof JsonArray) {
+            final JsonArray ja = (JsonArray)value;
+            return new AbstractList<Object>() {
+
+                @Override
+                public int size() {
+                    return ja.size();
+                }
+
+                @Override
+                public Object get(int i) {
+                    return ja.getValue(i);
+                }
+            };
+        }
+        if (value instanceof JsonObject) {
+            final JsonObject jo=(JsonObject)value;
+            return new AbstractList<Object>() {
```

Comment:
Poof!  It's gone!

Reply:
And now it's back, but nearly as short as the JsonArray accessor.  The objective here is to get O(1) performance when only size is needed.  In tests, size gets called 389 times and get is called  221 times, unrolling 81 distinct lists, making me suspect O(1) performance for size calls isn't a bad idea.

Reply:
I think you're on to something there with the `size` calls. It looks like [`BaseRuntime#deepEqualsArray`](https://github.com/burtcorp/jmespath-java/blob/5fa7ecadefa6dc720cd96bfe50da7cb4fb6d26fa/jmespath-core/src/main/java/io/burt/jmespath/BaseRuntime.java#L99), [`BaseRuntime#deepEqualsObject`](https://github.com/burtcorp/jmespath-java/blob/5fa7ecadefa6dc720cd96bfe50da7cb4fb6d26fa/jmespath-core/src/main/java/io/burt/jmespath/BaseRuntime.java#L113), and [`LengthFunction`](https://github.com/burtcorp/jmespath-java/blob/master/jmespath-core/src/main/java/io/burt/jmespath/function/LengthFunction.java#L19) all use `toList` to find the size of an array or object without (always) iterating it – the latter never iterates, the former only if the sizes match.

Perhaps there should be a runtime method for finding the size of an array or object? It would be an optimization for comparisons and the `length` function, and shouldn't be complicated for runtimes to support.